src/plugins/discover/public/application/components/context_app/context_app_legacy.tsx

```diff
@@ -49,9 +61,40 @@ function isLoading(status: string) {
 }
 
 export function ContextAppLegacy(renderProps: ContextAppProps) {
-  const status = renderProps.status;
-  const isLoaded = status === LoadingStatus.LOADED;
-  const isFailed = status === LoadingStatus.FAILED;
+  const services = getServices();
+  const { uiSettings: config, capabilities, indexPatterns } = services;
+  const {
+    indexPattern,
+    anchorStatus,
+    predecessorStatus,
+    successorStatus,
+    appState,
+    stateContainer,
+    hits: rows,
+    sorting,
+    filter,
+    minimumVisibleRows,
+    useNewFieldsApi,
+  } = renderProps;
+  const [expandedDoc, setExpandedDoc] = useState<EsHitRecord | undefined>(undefined);
+  const isAnchorLoaded = anchorStatus === LoadingStatus.LOADED;
+  const isFailed = anchorStatus === LoadingStatus.FAILED;
+  const areAnyDocsLoading =
+    anchorStatus === LoadingStatus.LOADING ||
+    predecessorStatus === LoadingStatus.LOADING ||
+    successorStatus === LoadingStatus.LOADING;
+  const isLegacy = config.get(DOC_TABLE_LEGACY);
```

Comment:
I think for the EuiDataGrid loading it is sufficient to take the status of the anchor loading into consideration. It's loaded first, so it's displayed, after that predecessor and successor are loaded. which is indicated by the spinners on top and bottom